### PR TITLE
feat(CDM-307): new code field skeleton and typescript migration

### PR DESCRIPTION
**Pull request by Nigui**
**What is the problem this PR is trying to solve?**

Current code field skeleton doesn't reflect content. 

image

**What is the chosen solution to this problem?**

- Create a new code skeleton to reflect code block. It uses design-system skeletons and tokens

image

- Migrate Code component to typescript and tests to RTL.
- Fix code storybook adding babelrc.json 

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/forms/package.json [hunk 1]
```diff
@@ -60,7 +61,6 @@
     "@storybook/addon-actions": "^6.5.9",
     "@storybook/addon-controls": "^6.5.9",
     "@talend/bootstrap-theme": "^7.0.0",
-    "@talend/design-system": "^7.1.6",
```
Comment (jmfrancois) [inline packages/forms/package.json:63]: you should in this case remove it from the peers

File: packages/forms/src/UIForm/fields/Code/displayMode/TextMode.component.test.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	const schema = {
+		title: 'My input title',
+	};
+
+	it('should render textarea', () => {
+		// when
+		const { baseElement } = render(<TextMode id="myForm" schema={schema} value="toto" />);
+
+		// then
+		expect(baseElement).toMatchSnapshot();
```
Comment (jmfrancois) [inline packages/forms/src/UIForm/fields/Code/displayMode/TextMode.component.test.tsx:17]: this is just a snapshot, and I don't see any textarea in it.

Comment (Nigui) [reply packages/forms/src/UIForm/fields/Code/displayMode/TextMode.component.test.tsx]: done, I missed to update test label.
I've removed snapshots

**Review by jmfrancois — APPROVED** [review]

File: packages/forms/package.json [hunk 3]
```diff
@@ -42,6 +42,7 @@
     "@talend/assets-api": "^1.2.1",
     "@talend/json-schema-form-core": "1.1.1",
     "@talend/react-components": "^9.2.0",
+    "@talend/design-system": "^7.1.6",
```
Comment (jmainguytalend) [inline packages/forms/package.json:45]: Are we sure we want to force host app to depend on multiple DS's version ?

Comment (Nigui) [reply packages/forms/package.json:45]: isn't already what we're doing?

Comment (jmainguytalend) [reply packages/forms/package.json:45]: yes, that i was checking 😅 
so if no problem with that...

**Review by jmainguytalend — APPROVED** [review]

**Pull request merged**